README.md

```diff
@@ -169,3 +169,37 @@ jobs:
           PR_URL: ${{github.event.pull_request.html_url}}
           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
 ```
+
+## Notes for project maintainers:
+
+<details><summary>:book: Release guide</summary>
+<p>
+
+  - Dependabot PR's:
+    - We expect Dependabot PRs to be passing CI and have any changes to the `dist/` folder built for production dependencies
+    - Some development dependencies may fail the `dist/` check if they modify the Typescript compilation, these should be updated manually
+  - Checkout and update `main` locally, then generate a patch release branch
+      ```bash
+      git checkout main
+      git pull
+      bin/bump-version -p patch
+      ```
+  - Generate a draft release for your new version
+      ```bash
+      gh release create v1.X.X --generate-notes --draft
+      > https://github.com/dependabot/fetch-metadata/releases/tag/untagged-XXXXXX
+      ```
+  - Create a PR linking to the release notes for review
+      ```bash
+      gh pr create --title "v1.X.X Release Notes" --body "https://github.com/dependabot/fetch-metadata/releases/tag/untagged-XXXXXX"
+      ```
+  - Get the PR reviewed, merge it and publish the release
+  - Update the `v1` tracking tag to point to the new version
+      ```bash
+      git pull
+      git checkout v1.x.x # Check out the release tag
+      git tag -f v1 # Force update the tracking tag
+      git push -f --tags
```

Comment:
I'm curious about why we need to do all the above. In my opinion, a tag should be fixed and not move. Forcing a tag to behave like a branch feels confusing. I understand that this is actually standard practice for handling many GitHub actions releases, but I'm curious about why. I have a feeling this practice may actually be widespread because of issues with Dependabot?

To clarify how I would expect Dependabot to work, I think someone using `dependabot/fetch-metadata@v1` in their workflow files should always get the latest v1.x.x release (tag) of the action, even if there's no tag named "v1". If this is how dependabot works, I think the better solution could be to actually delete the v1 and not run any of the four commands above when releasing.

Reply:
> In my opinion, a tag should be fixed and not move.

I understand this concern, but in this case we're conforming to the ecosystem's convention. GitHub has docs about [release management for GitHub Actions](https://docs.github.com/en/actions/creating-actions/about-custom-actions#good-practices-for-release-management), to which these instructions align.

To add some more context, here are the tags for one of the more popular Actions (`actions/checkout`):
<img width="896" alt="image" src="https://user-images.githubusercontent.com/5278382/191017554-7bccb855-628e-415f-962c-5eb3b2f598c2.png">

Reply:
Hi @landongrindheim 👋!

Thanks for pointing me to the official documentation, I knew this was standard practice, but I didn't know where this standard practice came from. It still doesn't make sense to me though.

Other actions that don't do this get the same end user behavior as far as I know, for example, https://github.com/ruby/setup-ruby. Users specifying `uses: ruby/setup-ruby@v1` also get the latest v1 release of the action, even if the action does not have a tag named "v1" always pointing to the latest v1 release. So I don't get the motivation for recommending this way of managing releases, in particular,

> Using this approach, your users can easily distinguish between major and minor versions

Am I missing something?

Reply:
@deivid-rodriguez I'm 100% certain you've thought more deeply about versioning than I have, so I don't be able to answer any questions re: whether this is a good approach 😅 I only know of this being a practice.

---

I looked at `ruby/setup-ruby` to see what was going on and saw that `v1` is a branch:

<img width="1549" alt="image" src="https://user-images.githubusercontent.com/5278382/192546810-2da85f31-e1ba-4bd8-b746-6e32b01f626c.png">

They don't have a `v1` tag: 

<img width="964" alt="image" src="https://user-images.githubusercontent.com/5278382/192547093-e965d71b-3a7a-4e7c-960f-09eb21be4f74.png">

I don't know if this is a recommended practice, but I kinda like that their tags are immutable at least 🙂

Reply:
Aaaaaah I missed that!

So, if `v1` was not a branch nor a tag, then `uses: ruby/setup-ruby@v1` would not work at all and raise an error, right? If that's the case, then that perfectly explains why we need to do this 😃!

It's true that `ruby/setup-ruby` tags are immutable, but it's also true that they are naming a branch using a name you usually expect to refer to a tag. So also confusing, although I like it better than the recommended approach of having moving tags.

I actually expected Github Actions to handle this "properly": if `v1` is not a tag nor a branch, but looks like a version, then automatically find the closest (semver wise) existing tag and use that.

Maybe I'll request this feature to Github Actions but I think my question is solved now, thanks @landongrindheim!

Reply:
I'm going to temp unresolve this as a reminder to myself to read up on this... and possibly raise the idea of converting `v1` to be a branch rather than a tag. I prefer immutable tags. I copied these instructions from one of our internal docs and didn't actually read them so didn't even realize this was being done this way/recommended til @deivid-rodriguez pointed out it.

Reply:
Just documenting what I found when I was going to request this feature.

* Funny enough, the maintainer of `ruby/setup-ruby` already requested to change the official recommendation from moving tags to moving branches: https://github.com/actions/toolkit/pull/631. But [a similar request](https://github.com/actions/toolkit/issues/214) was already closed, so it doesn't seem like it will be changed.

* The, in my opinion, best solution since it does not require moving tags nor branches, is already requested at https://github.com/orgs/community/discussions/9847. I thumbed up and subscribed to see if there's some movement there.

Reply:
After reading through the links, I dislike both tags and branches... 😀 

I think right now we should stick with what the ecosystem standard is, which is _generally_ moving a `vX` tag... If that later changes, we can easily update this here as well.

Reply:
Absolutely, my latest comments were not meant at blocking this PR!

Reply:
Cool. And FWIW, thanks for posting those links, I also subscribed, and the discussion was educational, I hadn't thought about the downsides tradeoffs of using a branch before this.